llvm/CMakeLists.txt

```diff
@@ -673,6 +673,9 @@ option(LLVM_USE_OPROFILE
 option(LLVM_EXTERNALIZE_DEBUGINFO
   "Generate dSYM files and strip executables and libraries (Darwin Only)" OFF)
 
+option(LLVM_ENABLE_EXPORTED_SYMBOLS
```

Comment:
We may want to rename this. The name hints that this option applies to everything but actually it only applies to executables and not libraries. E.g. `LLVM_ENABLE_TOOLS_WITH_EXPORTED_SYMBOLS`

Reply:
"tools" has a pretty specific meaning for LLVM so I think that will actually cause more confusion (i.e. I would expect that option to apply to `llvm-dwarfdump`, but not to `clang`). I'm personally fine with the current concise name.

Reply:
Would `LLVM_ENABLE_EXECUTABLES_WITH_EXPORTED_SYMBOLS` be too long?

Reply:
I'm fine with that. You could also do `LLVM_ENABLE_EXPORTED_SYMBOLS_IN_EXECUTABLES` which is ever so slightly shorter 🤷‍♂️

Reply:
> "tools" has a pretty specific meaning for LLVM so I think that will actually cause more confusion (i.e. I would expect that option to apply to llvm-dwarfdump, but not to clang). I'm personally fine with the current concise name.

I'm not exactly sure what that "specific meaning" is but I'm happy to go with something else. I think using the name "EXECUTABLE" as @cyndyishida is suggesting is consistent with the option being used in `add_llvm_executable()`.

Reply:
@JDevlieghere 
> "tools" has a pretty specific meaning for LLVM so I think that will actually cause more confusion (i.e. I would expect that option to apply to llvm-dwarfdump, but not to clang). I'm personally fine with the current concise name.

I'm not exactly sure what that "specific meaning" is but I'm happy to go with something else. I think using the name "EXECUTABLE" as @cyndyishida is suggesting is consistent with the option being used in the `add_llvm_executable()` macro.

Reply:
@delcypher I was referring to the tools in `/llvm-project/llvm/tools`, which use `add_llvm_tool_subdirectory` and `add_llvm_tool` but maybe the distinction is more murky than I think it is. I'm also fine with `EXECUTABLE` in the name 👍